internal/thea.go

```diff
@@ -3,323 +3,196 @@ package internal
-	"github.com/hbomb79/Thea/pkg/worker"
 )
 
-var log = logger.Get("Thea")
+var log = logger.Get("Core")
 
-// Thea exposes the core workflow for the processor. There are two main categories of methods exposed here:
-//
-// -- Service Layer APIs --
-// These are the preferred way to interact with Thea. Methods enclosed within these APIs are aware of all state
-// in the Thea runtime and will ensure that updates are applied correctly across all of it (e.g. cancelling an
-// item may remove it from the queue, and cancel all ffmpeg actions, and send an update to the client). Each API
-// here is a "service" as it encapsulates related behaviour - however a call to one service may incur calls to other
-// services via their respective API as well (these "side-effects" after often required in order to ensure Theas state is
-// kept valid!).
-//
-// -- Internal APIs --
-// Each of these internal APIs represent a single "unit" or component of the Thea state - updating one of these states without
-// understanding the implications that may have on the other units is dangerous! (This is why the above service
-// layer APIs are preferred). These internal APIs allow code that knows what it's doing to selectively change the
-// state of Thea without unindented side-effects (however these side-effects are often a good thing!).
-type Thea interface {
-	UpdateManager
-	CoreService
-	ProfileService
-	QueueService
-	MovieService
-
-	Start() error
-	Stop()
-
-	queue() queue.QueueManager
-	ffmpeg() ffmpeg.FfmpegCommander
-	profiles() profile.ProfileManager
-	workerPool() *worker.WorkerPool
-	config() TheaConfig
-}
+const (
+	THEA_USER_DIR_SUFFIX = "/thea/"
+)
 
-// Thea represents the top-level object for the server, and is responsible
-// for initialising embedded support services, workers, threads, event
-// handling, et cetera...
-type theaImpl struct {
-	UpdateManager
-	CoreService
-	ProfileService
-	QueueService
-	MovieService
+type (
+	EventParticipator interface {
+		RegisterEventCoordinator(event.EventCoordinator)
+	}
 
-	queueMgr   queue.QueueManager
-	ffmpegMgr  ffmpeg.FfmpegCommander
-	profileMgr profile.ProfileManager
-	workers    *worker.WorkerPool
+	RunnableService interface {
+		Run(context.Context) error
+	}
 
-	cfg               TheaConfig
-	theaCtx           context.Context
-	theaCtxCancel     context.CancelFunc
-	shutdownWaitGroup *sync.WaitGroup
-}
+	RestGateway interface {
+		RunnableService
+		BroadcastTaskUpdate(uuid.UUID) error
+		BroadcastTaskProgressUpdate(uuid.UUID) error
+		BroadcastWorkflowUpdate(uuid.UUID) error
+		BroadcastMediaUpdate(uuid.UUID) error
+		BroadcastIngestUpdate(uuid.UUID) error
+	}
 
-const THEA_USER_DIR_SUFFIX = "/thea/"
-const THEA_CACHE_FILE_PATH = "cache.json"
-const THEA_PROFILE_FILE_PATH = "profiles.json"
-const THEA_UPDATE_INTERVAL = time.Second * 2
-const THEA_QUEUE_SYNC_INTERVAL = time.Second * 5
+	TranscodeService interface {
+		RunnableService
+		EventParticipator
+		NewTask(uuid.UUID, uuid.UUID) error
```

Comment:
Name these arguments to help document which UUID is which.

Reply:
Good point, demonstrated quite well when I went to try and name them hahaha